src/components/modals/send-modal/index.js

```diff
@@ -39,12 +45,39 @@ const SendModal = ({ show, handleModal, utxo, onSale }) => {
         return key.length === 33 ? key.slice(1, 33) : key;
     }
 
+    // sign message with first sign transaction
+    const TAPROOT_MESSAGE = (domain) =>
+    // will switch to nosft.xyz once sends are implemented
+    `Sign this message to generate your Bitcoin Taproot key. This key will be used for your ${domain} transactions.`;
+
     async function sendUtxo() {
         const inputAddressInfo = getAddressInfo(nostrPublicKey);
+        const domain = SessionStorage.get(SessionsStorageKeys.DOMAIN);
         const psbt = new bitcoin.Psbt({
             network: TESTNET ? bitcoin.networks.testnet : bitcoin.networks.bitcoin,
         });
-        const publicKey = Buffer.from(await window.nostr.getPublicKey(), "hex");
+        const { ethereum } = window;
+        const ethAddress = ethereum.selectedAddress;
```

Comment:
if ethereum is undefined, would this break the code?

Reply:
Nope, still works ok